File: backend/models/quizz.js
const mongoose = require("mongoose");

const QuizzSchema = new mongoose.Schema({
  question: {
    type: String,
    required: true,
  },
  choice: {
    type: Array,
    required: true,
  },
  answer: {
    type: Array,
    required: true,
  },
});

module.exports = mongoose.model("Quizz", QuizzSchema);

File: mobile/lib/data/models/quizz.dart
import 'package:uit_elearning/data/models/base_model.dart';

class Quiz extends BaseModel {
  final String question;
  final List<String> choices;
  final List<int> answers;

  Quiz({
    String? id,
    required this.question,
    required this.choices,
    required this.answers,
  }) : super(id);

  factory Quiz.fromJson(Map<String, dynamic> json) => Quiz(
        id: json["_id"],
        question: json["question"],
        choices: List<String>.from(json["choice"].map((x) => x)),
        answers: List<int>.from(json["answer"].map((x) => x)),
      );
}
